homeassistant/components/roborock/coordinator.py

```diff
@@ -222,10 +222,21 @@ async def _async_update_data(self) -> DeviceState:
             # since the last map update, you can update the map.
             new_status = self.properties_api.status
             if (
-                new_status.in_cleaning
-                and (dt_util.utcnow() - self._last_home_update_attempt)
-                > IMAGE_CACHE_INTERVAL
-            ) or self.last_update_state != new_status.state_name:
+                (
+                    new_status.in_cleaning
+                    and (dt_util.utcnow() - self._last_home_update_attempt)
+                    > IMAGE_CACHE_INTERVAL
+                )
+                or self.last_update_state != new_status.state_name
+                or (
```

Comment:
I worry this will make it run faster than IMAGE_CACHE_INTERVAL when map is None (e.g. if its broken and stuck as None)